src/mdast-process-images.js

```diff
@@ -44,17 +64,31 @@ export async function processImages(log, tree, mediaHandler, baseUrl) {
         node.url = new URL(url, baseUrl).href;
         register(node);
       } else if (url.startsWith('https://')) {
+        if (isExternalImage(url, externalImages)) {
+          // Add custom class to the node for external images
+          const data = node.data || {};
+          const hProperties = data.hProperties || {};
+          hProperties.className = 'external-image';
+          // eslint-disable-next-line no-param-reassign
+          node.data = { ...data, hProperties };
+
+          // eslint-disable-next-line no-param-reassign
+          node.data.externalImage = true;
+
+          log.debug(`Marked external image: ${url}`);
+        }
+
         register(node);
       }
     }
     return CONTINUE;
   });
 
-  if (images.size > 200) {
-    throw new TooManyImagesError(`maximum number of images reached: ${images.size} of 200 max.`);
+  if (images.size + externalImageNodes.size > 200) {
+    throw new TooManyImagesError(`maximum number of images reached: ${images.size + externalImageNodes.size} of 200 max.`);
```

Comment:
we don't care if the document has too many external images. no need to include them in the check.

Reply:
done